src/classic/clvm_tools/stages/stage_2/optimize.rs

```diff
@@ -24,11 +30,38 @@ use crate::classic::clvm_tools::NodePath::NodePath;
 
 use crate::util::{number_from_u8, u8_from_number};
 
+use crate::compiler::sexp::decode_string;
+
 #[derive(Clone)]
 pub struct DoOptProg {
     runner: Rc<dyn TRunProgram>,
 }
 
+lazy_static! {
+    pub static ref DEPTH_COUNT: AtomicI32 = AtomicI32::new(0);
+}
+
+pub fn depth_count() -> i32 {
+    DEPTH_COUNT.fetch_add(0, Ordering::SeqCst)
+}
+
+pub fn enter() {
+    DEPTH_COUNT.fetch_add(1, Ordering::SeqCst);
+}
+
+pub fn leave() {
+    DEPTH_COUNT.fetch_add(-1, Ordering::SeqCst);
+}
+
+pub fn indent() -> String {
+    let dc = depth_count() as usize;
+    let mut v = Vec::with_capacity(dc);
```

Comment:
you can call `push()` on a `std::string::String` also, I don't think you need to create a vector first

Reply:
this was all removed